### PR TITLE
Add command to open legacy notebook.

### PR DESCRIPTION
**Pull request by jasongrout**
I put it under the help menu - suggestions for a better place are welcome.

@blink1073 - is there a more sanctioned place to get the baseurl?

### CODE REVIEW

**Comment by jasongrout** [comment]
CC @fperez, who asked for this at some point.

**Comment by fperez** [comment]
I don't actually recall asking for this, but +1 anyways :)

**Comment by blink1073** [comment]
@jasongrout, technically it should be provided by the `services` plugin, but it would require adding `baseUrl` to `IServiceManager` in `jupyter-js-services`: https://github.com/jupyter/jupyter-js-services/blob/master/src/manager.ts

**Comment by damianavila** [comment]
Are we sure we want to open the legacy notebook from the JupyterLab UI? I am not sure about that... in fact, in a adoption push-thing practice, I would like people to actually "do work" if they really want to use the legacy notebook, so they keep using the new one... "helping" them with with the migration.

**Comment by jasongrout** [comment]
I didn't think so at first, but there have been a number of situations recently where we wanted people to check something for debugging in the current notebook, and it would have been easiest to tell them to just open a menu item and check it.

Fernando also brought up that it's nice to have them both open side by side for comparison.

Also, it *is* in our migration plan that the current notebook becomes an optional plugin when JLab becomes default.

**Comment by damianavila** [comment]
>I didn't think so at first, but there have been a number of situations recently where we wanted people to check something for debugging in the current notebook, and it would have been easiest to tell them to just open a menu item and check it.

Good point...

>Also, it is in our migration plan that the current notebook becomes an optional plugin when JLab becomes default.

Yep, I know... but since it will be an "optional" plugging, I am not sure to have a default link to open it...
But I guess I am OK to have it for now, in the early transition step... later is debatable :wink:

**Comment by SylvainCorlay** [comment]
I am 👎 on this.

If the goal is to separate the `notebook` repo into an independent jupyter server and make the legacy notebook frontend a serverextension just like jupyterlab, this creates a dependency of jupyterlab to the presence of the legacy frontend.

**Comment by blink1073** [comment]
The classic notebook would have to be heavily refactored in order to exist as a pane in JupyterLab, because it is making a lot of assumptions about being the only thing on the page (e.g. using node ids).   The "plugin" could be a way to launch a new browser tab for a notebook file using the classic notebook.

**Comment by SylvainCorlay** [comment]
I never suggested to make it exist as a pane. The legacy notebook should be a serverextension that adds handlers for the /tree and /notebooks urls just like jupyterlab adds handlers for /lab.

**Comment by blink1073** [comment]
I wasn't addressing your comment per se @SylvainCorlay, just stating my thoughts on the subject overall.

**Comment by jasongrout** [comment]
> Yep, I know... but since it will be an "optional" plugging, I am not sure to have a default link to open it...
> But I guess I am OK to have it for now, in the early transition step... later is debatable 😉

I fully expect this to be optional in the future.

**Comment by jasongrout** [comment]
> The "plugin" could be a way to launch a new browser tab for a notebook file using the classic notebook.

Good point - "Open with... classic notebook"

**Comment by jasongrout** [comment]
> If the goal is to separate the notebook repo into an independent jupyter server and make the legacy notebook frontend a serverextension just like jupyterlab, this creates a dependency of jupyterlab to the legacy frontend package.

During this alpha/beta phase, I still think a link like this is very handy for people to quickly compare the old and new notebook frontends. The problem with saying "just launch `jupyter notebook`" is that it launches a separate notebook server, so you don't have access to the lab running kernels, etc. I've been asking people a lot lately to change the url to `/tree` to check if something works in the old notebook for debugging. This menu item makes it easier for someone to do so.

**Comment by ellisonbg** [comment]
* In the long run we won't need this...
* For now, I think this is very important to provide a seamless experience for users...

**Comment by ellisonbg** [comment]
Can we implement this in a slightly different manner:

Add a context menu item to the filebrowser that says "Open in classic notebook"

Much more discoverable and useful for users maybe?

File: src/help/plugin.ts [hunk 1]
```diff
@@ -140,6 +144,15 @@ function activateHelpHandler(app: JupyterLab, mainMenu: IMainMenu, palette: ICom
     category: 'Help'
   }));
 
+  let openLegacyNotebookId = 'legacy-notebook:open';
+  app.commands.addCommand(openLegacyNotebookId, {
+    label: 'Legacy Jupyter Notebook',
```
Comment (ellisonbg) [inline src/help/plugin.ts]: I tend to like "classic" rather than "legacy"...

Would also start the command with a verb: "Open Classic Notebook"

**Comment by jasongrout** [comment]
> Add a context menu item to the filebrowser that says "Open in classic notebook"

I think there still is value to opening `/tree`, as well as individual notebooks.

File: src/help/plugin.ts (hunk 1 above)
Comment (jasongrout) [inline src/help/plugin.ts]: Makes sense; I'll change the wording.

**Comment by ellisonbg** [comment]
True, probably makes sense to have a global "Open Classic notebook" as well. We can add the per document open in a separate PR.

File: src/help/plugin.ts (hunk 1 above)
Comment (jasongrout) [inline src/help/plugin.ts]: I did think `legacy` was better than `deprecated` or `decrepit` :)

**Pull request merged**